### ISSUE TITLE
Unreachable case involving path dependent types

### ISSUE THREAD

**Issue by bishabosha** [lampepfl/dotty#12681]
## Compiler version

3.0.0

## Minimized code

```Scala
object Examples {

  case class Leaf1() extends i.Root
  case class Leaf2() extends i.Branch

  val i = new Inner()

  class Inner {

    sealed trait Root
    sealed trait Branch extends Root

    // simulate ordinal method of a Mirror.SumOf generated at this call site
    def myOrdinal(r: Root): Int = r match {
      case _: Examples.Leaf1 => 0
      case _: Inner.this.Branch => 1
    }
  }

}
```

## Output

```scala
-- [E030] Match case Unreachable Warning: example3.scala:16:11 -----------------
16 |      case _: Inner.this.Branch => 1
   |           ^^^^^^^^^^^^^^^^^^^^
   |           Unreachable case
1 warning found
```

## Expectation

in Scala 2.13.6 we get the type error

```scala
cmd0.sc:14: pattern type is incompatible with expected type;
 found   : ammonite.$sess.cmd0.Examples.Leaf1
 required: Inner.this.Root
      case _: Examples.Leaf1 => 0
                       ^
Compilation Failed
```

**Comment by bishabosha** [comment]
the unreachable case warning seems confusing because it only appears when the `Leaf1` branch is above it, which is unreleated

**Comment by liufengyun** [comment]
> the unreachable case warning seems confusing because it only appears when the `Leaf1` case is above it, which is unrelated

The problem is related to how we select qualified children of a sealed type. We need to tweak the subtyping so that `Leaf2` is a qualified child of `Inner.this.Branch` despite that `Leaf2` is not a subtype of `Inner.this.Branch`. I'll propose a fix.

**Comment by liufengyun** [comment]
Fixed as part of #12549.
